README.md

```diff
@@ -4,10 +4,43 @@ The Cloud Armor Node Controller can help provide UDP DDoS protection for game cu
+    ```shell
+    export REPOSITORY=<REPOSITORY>
+    docker build --tag=$REPOSITORY/cloud-armor-node-controller:0.1
+    docker push $REPOSITORY/cloud-armor-node-controller:0.1
+    ```
 
-2. Push the docker image to your desired Artifact Registry repository by running `docker push $REPOSITORY/cloud-armor-node-controller:0.1`.
+    Or alternatively with Cloud Build:
+    ```shell
+    export REPOSITORY=<REPOSITORY>
+    make cloud-build REPOSITORY=$REPOSITORY
+    ```
 
-3. Give IAM roles `roles/compute.instanceAdmin.v1` and `roles/compute.networkAdmin` to Kubernetes Service Account `cloud-armor-node-controller-sa` (The full name of the KSA should be in the format of `principal://iam.googleapis.com/projects/PROJECT_ID/locations/global/workloadIdentityPools/PROJECT_ID.svc.id.goog/subject/ns/cloud-armor-node-controller/sa/cloud-armor-node-controller-sa`).
+2. Give IAM roles `roles/compute.instanceAdmin.v1` and `roles/compute.networkAdmin` to Kubernetes Service Account `cloud-armor-node-controller-sa`:
+    ```shell
+    export PROJECT_ID=<PROJECT_ID>
+    export PROJECT_NUMBER=<PROJECT_NUMBER>
+    export KUBERNETES_SERVICE_ACCOUNT=principal://iam.googleapis.com/projects/$PROJECT_NUMBER/locations/global/workloadIdentityPools/$PROJECT_ID.svc.id.goog/subject/ns/cloud-armor-node-controller/sa/cloud-armor-node-controller-sa
 
-3. Deploy the Cloud Armor Node Controller in your cluster in `cloud-armor-node-controller` namespace by running `kubectl apply -f deployment.yaml`.
+    gcloud projects add-iam-policy-binding projects/$PROJECT_ID --role=roles/compute.instanceAdmin.v1 --member=$KUBERNETES_SERVICE_ACCOUNT --condition=None
+    gcloud projects add-iam-policy-binding projects/$PROJECT_ID --role=roles/compute.networkAdmin --member=$KUBERNETES_SERVICE_ACCOUNT --condition=None
+    ```
+
+3. To deploy, edit the `deployment.yaml` file: 
+    * Replace `YOUR_REPOSITORY_HERE` with the repository chosen above.
+    * Replace `SELECTOR` to be a Kubernetes label selector, for example - `cloud.google.com/gke-nodepool=default-pool`.
+    * Replace `SECURITY_POLICY` to be the name of a security policy you created(Note: just the name, not the full path). 
+    
+    Then, deploy the Cloud Armor Node Controller in your cluster in `cloud-armor-node-controller` namespace with:
+    ```shell
+    kubectl apply -f deployment.yaml
+    ```
+
+## Notes
+
+* The Cloud Armor Node Controller will set the security policy of the nodes that are selected by the `SELECTOR` enironment variable **exactly once**. More specifically, it will set the security policy the first time it sees a node create/update event. If later the security policy on the node is changed by some other means, such as with `gcloud` commands, then the Cloud Armor Node Controller will not overwrite the security policy immediately; instead, it will overwrite the security policy after the pod is rescheduled.
```

Comment:
Maybe instead of `after the pod is rescheduled`, something like `instead, it will overwrite the security policy if the Cloud Armor Node Controller is restarted, such as when a pod is rescheduled`.

But also I'm not sure about emphasizing `exactly once` when that condition exists?

Reply:
Agreed